src/gui/PatternsManagerWindow.cpp

```diff
@@ -54,6 +54,38 @@ void PatternsManagerWindow::create_pattern()
     if (!ok || pat_name.isEmpty())
         return;
 
+    const auto new_pfs_name = to_u8string(pat_name);
+
+    // Check whether such pattern already exists.
+    auto all_pfs = settings_.current_profile().per_file_settings();
+    auto found   = std::ranges::find(all_pfs, new_pfs_name, [](const PerFileSettings *pfs) {
+        return pfs->pattern.text();
+    });
+
+    if (found != all_pfs.end())
+        return;
+
+    //Choosing Pattern type
+
+    QStringList type_list{"Glob", "Regex"};
+    const QString type_str = QInputDialog::getItem(this,
+                                                   tr("Pattern type"),
+                                                   tr("What kind of a pattern would you like to create?"),
+                                                   type_list,
+                                                   0,
+                                                   false,
+                                                   &ok);
+    if (!ok)
+        return;
+
+    Pattern::Type pattern_type;
+    if (type_str == "Glob")
+        pattern_type = Pattern::Type::Wildcard;
+    else if (type_str == "Regex")
+        pattern_type = Pattern::Type::Regex;
+    else
+        return;
```

Comment:
This could be
```suggestion
    const auto pattern_type = nlohmann::json{type_str}.get<Pattern::Type>();
```
Not sure if this is better

Reply:
That actually doesn't compile,

```
Severity	Code	Description	Project	File	Line	Suppression State	Details
Error	C2440	'<function-style-cast>': cannot convert from 'initializer list' to 'nlohmann::json_abi_v3_11_3::json' 	C:\Dev\VS\cathedral-assets-optimizer\build\cathedral-assets-optimizer	C:\Dev\VS\cathedral-assets-optimizer\src\gui\PatternsManagerWindow.cpp	81		
```

Reply:
Oh, yes, probably because it's a QString. Don't bother